### ISSUE TITLE
Data and index compression

### ISSUE THREAD

**Issue by Ryba1986** [dotnet/efcore#26049]
Does EF Core support table and index compression in model building or ?

```sql
CREATE TABLE [table] (...) WITH (DATA_COMPRESSION = PAGE);
```

**Comment by roji** [comment]
No, this isn't currently supported. However, you should be able to edit your migration and use [raw SQL](https://docs.microsoft.com/en-us/ef/core/managing-schemas/migrations/managing?tabs=dotnet-core-cli#adding-raw-sql) to do ALTER TABLE to turn on compression after the table is created.

**Comment by clement911** [comment]
We would also use such a feature if available. 

We try to avoid changing generated migrations so that new databases can be created from scratch for dev or sharding or db migration purposes,

This would also be useful on the primary key (.HasKey).

Supporting all the index creation options via fluent syntax might be a lot.
Maybe a better alternative is to be able to specific a string of options that should be included in the WITH clause.

e.g.
`builder.Entity<SaleTransaction>().HasKey(i => i.Id).With("DATA_COMPRESSION = PAGE, FILLFACTOR = 90, PAD_INDEX = ON")`
